enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,267 @@
+---
+title: prometheus-alerts-for-insights-recommendations
+authors:
+  - "@natiiix"
+reviewers:
+  - "@tremes"
+  - "@inecas"
+approvers:
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+creation-date: 2022-02-14
+last-updated: 2022-02-17
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/CCXDEV-6653
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insights Operator Prometheus Alerts for Insights Recommendations
+
+## Summary
+
+Insights recommendations are triggered based on data in the archives
+periodically uploaded by the Insights Operator. A report about these
+recommendations is available through the Insights Smart Proxy, and it is
+currently fetched by the Insights Operator after each successful archive
+upload in order for the Insights Operator to update Prometheus metrics
+containing the number of currently active recommendations in each severity
+category.
+
+This enhancement aims to extend this functionality by not only making the
+number of recommendations available, but also providing specific information
+about each of the recommendations via Prometheus alerts. The currently
+suggested approach expects one info-level alert per active Insights
+recommendation, containing its ID, human-readable name, severity, and a link
+to detailed information about the recommendation and its remediation steps.
+
+## Motivation
+
+Users are currently not being immediately notified about newly active Insights
+recommendations. If they do not check the Insights tab periodically, these
+recommendations may go unnoticed for a long time during which the underlying
+issue escalates instead of being resolved. Making the recommendations available
+through the Prometheus alerts makes it easier for users to define how they want
+to receive information about newly active recommendations (via an email, Slack
+message, etc.).
+
+### Goals
+
+- New Prometheus metric with labels containing aforementioned properties of
+  the recommendation that will indicate if the recommendation is currently active.
+- New Prometheus alert based on this metric, defined using a template that fills
+  the information from labels into the description of the alert.
+- The user should be able to find out when was a certain recommendation active
+  by querying the Prometheus metric (and plot this in Grafana, for example).
+- Configuration option allowing the user to disable/enable this functionality.
```

Comment:
Maybe I missed that but I think we should explicitly mention that it will be opt-in feature (disabled by default)

Reply:
I don't think we want to backport it. I guess enabling it by default will introduce a couple of new alerts, which someone (SRE?) might find a little bit annoying (even though they will be only info alerts). I think the motivation for enabling it is to prefer to use the alertmanager (and its own notification configuration) as opposed to the "common" Insights advisor.

Reply:
given the info state - potentially combined with a custom label for explicit identification? SRE should have no issue with this as we can route on sev + label accordingly.